sdk/storage/azure-storage-blob/azure/storage/blob/_shared/policies_async.py

```diff
@@ -46,7 +46,10 @@ async def is_checksum_retry(response):
     # retry if invalid content md5
     if response.context.get('validate_content', False) and response.http_response.headers.get('content-md5'):
         try:
-            await response.http_response.read()  # Load the body in memory and close the socket
+            resp = response.http_response
+            # resp._decompress = False
```

Comment:
Note: We would never ship with something like this, but this illustrates that in the content validation path, it is respecting that `_decompress` value iirc